src/mscorlib/src/System/Globalization/TimeSpanParse.cs

```diff
@@ -157,71 +142,93 @@ public bool IsInvalidNumber(int maxValue, int maxPrecision)
         //
         //  Actions: TimeSpanTokenizer.GetNextToken() returns the next token in the input string.
         // 
+        [IsByRefLike]
         private struct TimeSpanTokenizer
         {
+            private ReadOnlySpan<char> m_value;
             private int m_pos;
-            private String m_value;
 
-            internal void Init(String input)
-            {
-                Init(input, 0);
-            }
-            internal void Init(String input, int startPosition)
+            internal TimeSpanTokenizer(ReadOnlySpan<char> input) : this(input, 0) { }
+
+            internal TimeSpanTokenizer(ReadOnlySpan<char> input, int startPosition)
             {
-                m_pos = startPosition;
                 m_value = input;
+                m_pos = startPosition;
             }
+
             // used by the parsing routines that operate on standard-formats
             internal TimeSpanToken GetNextToken()
             {
-                Debug.Assert(m_pos > -1);
-
-                TimeSpanToken tok = new TimeSpanToken();
-                char ch = CurrentChar;
-
-                if (ch == (char)0)
+                // Get the position of the next character to be processed.  If there is no
+                // next character, we're at the end.
+                int pos = m_pos;
+                Debug.Assert(pos > -1);
+                if (pos >= m_value.Length)
                 {
-                    tok.ttt = TTT.End;
-                    return tok;
+                    return new TimeSpanToken(TTT.End);
                 }
 
-                if (ch >= '0' && ch <= '9')
+                // Now retrieve that character. If it's a digit, we're processing a number.
+                int num = m_value[pos] - '0';
```

Comment:
nit: maybe we can simplify the code for the leading zeros here by just doing 

while (m_pos <= m_value.Length && m_value[m_pos] == '0')
    m_pos++;

if (m_pos >= m_value.Length || (uint)(m_value[m_pos] - '0') > 9)
   return new TimeSpanToken(TTT.Num, 0, m_pos - pos, default(ReadOnlySpan<char>));

The current code looks ok too.